### PR TITLE
Add no_std support

### PR DESCRIPTION
**Pull request by oberien**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,11 +13,13 @@ keywords = ["encryption", "xts"]
 exclude = ["test_files"]
 
 [features]
+default = ["std"]
+std = ["byteorder/std", "cipher/std"]
```
Comment (pheki) [inline Cargo.toml]: As we don't currently use their std features, can we just remove both from here?

```
std = []
```

File: Cargo.toml [hunk 2]
```diff
@@ -13,11 +13,13 @@ keywords = ["encryption", "xts"]
...
 openssl_tests = ["openssl"]
 
 [dependencies]
-cipher = "0.4"
-byteorder = "1"
+cipher = { version = "0.4", default-features = false }
```
Comment (pheki) [inline Cargo.toml]: I don't think we need to disable `default-features` as cipher doesn't enable `std` by default

**Review by pheki — CHANGES_REQUESTED** [review]
Thanks a lot! I just have 2 small suggestions.

**Comment by oberien** [comment]
Thanks for the review. I've fixed both remarks. Both `cargo test` and `cargo test --no-default-features` as well as compiling against uefi (the no-std target I'm working with) run without problems.

**Comment by pheki** [comment]
Tested and merged it.

Thanks again :)

**Pull request merged**